airbyte-integrations/connectors/source-typeform/source_typeform/source.py

```diff
@@ -0,0 +1,230 @@
+# SOFTWARE.
+#
+
+
+import urllib.parse as urlparse
+from abc import ABC, abstractmethod
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+from urllib.parse import parse_qs
+
+import pendulum
+import requests
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from pendulum.datetime import DateTime
+
+
+class TypeformStream(HttpStream, ABC):
+    url_base = "https://api.typeform.com"
+    # maximum number of entities in API response per single page
+    limit: int = 200
+    date_format: str = "YYYY-MM-DDTHH:mm:ss[Z]"
+
+    def __init__(self, **kwargs: Mapping[str, Any]):
+        super().__init__(authenticator=kwargs["authenticator"])
+        self.config: Mapping[str, Any] = kwargs
+        self.start_date: DateTime = pendulum.from_format(kwargs["start_date"], self.date_format)
+
+        # changes page limit, this param is using for development and debugging
+        if kwargs.get("page_size"):
+            self.limit = kwargs.get("page_size")
+
+    def next_page_token(self, response: requests.Response) -> Optional[Any]:
+        return None
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        yield from response.json()["items"]
+
+
+class TrimForms(TypeformStream):
+    """
+    This stream is responsible for fetching list of from_id(s) which required to process data from Forms and Responses.
+    API doc: https://developer.typeform.com/create/reference/retrieve-forms/
+    """
+
+    primary_key = "id"
+    limit: int = 200
+
+    def path(
+        self,
+        stream_state: Mapping[str, Any] = None,
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Optional[Any] = None,
+    ) -> str:
+        return "/forms"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Any]:
+        page = self.get_current_page_token(response.url)
+        # stop pagination if current page equals to total pages
+        return None if not page or response.json()["page_count"] <= page else page + 1
+
+    def get_current_page_token(self, url: str) -> Optional[int]:
+        """
+        Fetches page query parameter from URL
+        """
+        parsed = urlparse.urlparse(url)
+        page = parse_qs(parsed.query).get("page")
+        return int(page[0]) if page else None
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, any] = None,
+        next_page_token: Optional[Any] = None,
+    ) -> MutableMapping[str, Any]:
+        params = {"page_size": self.limit}
+        params["page"] = next_page_token or 1
+        return params
+
+
+class StreamMixin:
```

Comment:
Can we rename this to `TrimFormsMixin` to make it clear that anything relying on this mixin is actually executing code form the `TrimForms` to get the required data to retrieve it's own data.